app/models/user.rb

```diff
@@ -482,6 +485,42 @@ def title
     name
   end
 
+  def number_visits_per_day
+    return user.sign_in_count.round(1) if user.created_at.today?
+
+    (user.sign_in_count.to_f / (Time.zone.today - user.created_at.to_date).to_i).round(1)
+  end
+
+  def number_articles_read
+    user.page_views.select(:article_id).distinct.count
+  end
+
+  def number_articles_per_day
+    article_count = user.page_views.select(:article_id).distinct.count
```

Comment:
counting the number of distinct `articles` that correspond to the `page_views` associated with the current `user`